src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs

```diff
@@ -49,19 +47,17 @@ public RequestThrottlingMiddleware(RequestDelegate next, ILoggerFactory loggerFa
         /// <returns>A <see cref="Task"/> that completes when the request leaves.</returns>
         public async Task Invoke(HttpContext context)
         {
-            Interlocked.Increment(ref _queuedRequests);
+            var waitInQueueTask = _queuePolicy.TryEnterAsync();
 
-            var success = false;
-            try
-            {
-                success = await _queuePolicy.TryEnterAsync();
-            }
-            finally
+            if (!waitInQueueTask.IsCompleted)
             {
-                Interlocked.Decrement(ref _queuedRequests);
+                using (RequestThrottlingEventSource.Log.QueueTimer())
```

Comment:
If we're calculating "Average Queue Duration" or "Average Time in Queue" using this counter, shouldn't we count requests that don't need to wait on the queue as spending 0 time in the queue?

Reply:
Nice that's a big oversight from me. Want to either
 - add a "queue skipped" event that reports duration 0
 - wrap more of the middleware invocation in the `using(QueueTimer)`